src/nodes/predicate.ts

```diff
@@ -49,21 +50,35 @@ export class PredicateNode<t = unknown> extends Node<"predicate"> {
 
     static from<def extends PredicateNodeInput>(def: def) {
         const constraints: ConstraintNode[] = []
+        const basisNode = new BasisNode(def.basis)
         if (def.divisor) {
-            constraints.push(new DivisibilityNode(def.divisor))
+            basisNode.domain === "number"
```

Comment:
I usually just use an if statement in cases like this if I'm not returning anything or setting a value directly.